deployment/staging/deploy.sh

```
@@ -0,0 +1,149 @@
+run_service() {
+    local target=$1
+    log_info "Starting ${target^^} container"
+    ${DOCKER_COMPOSE} up -d "${target}" --scale "${target}=${MAX_SCALE}"
+}
+
+check_health_service() {
+    local target=$1
+    log_info "Performing health check for ${target^^}"
+
+    local retries=0
+    local container_ids=( $(${DOCKER_COMPOSE} ps -q "${target}") )
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)